rpcd/playbooks/roles/elasticsearch/tasks/upgrade_pre_flight.yml

```diff
@@ -0,0 +1,76 @@
+---
+# Copyright 2016, Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+- name: "Get ElasticSearch version"
+  command: "dpkg-query --show --showformat='{$Version}' elasticsearch"
+  register: elasticsearch_installed_version
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - elasticsearch-pre-flight
+
+- name: "Check for ElasticSearch version > 2.x"
+  set_fact:
+    proceed_with_upgrade: false
+    latest_version: true
+  register: proceed_with_upgrade
+  when: "{{ elasticsearch_installed_version.stdout | version_compare(elasticsearch_version, '>=', strict=True) }}"
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - elasticsearch-pre-flight
+
+- name: "Check for ElasticSearch version 1.7.x"
+  set_fact:
+    proceed_with_upgrade: false
+    reindex_version: true
+  register: proceed_with_upgrade
+  when:
+    - "{{ elasticsearch_installed_version.stdout | version_compare(elasticsearch_reindex_version, '>=', strict=True) }}"
+    - "{{ elasticsearch_installed_version.stdout | version_compare('1.7.9', '<=', strict=True) }}"
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - elasticsearch-pre-flight
+
+- name: "Check for ElasticSEarch version > 1.4.x"
+  set_fact:
+    proceed_with_upgrade: true
+  register: proceed_with_upgrade
+  when:
+    - "{{ elasticsearch_installed_version.stdout | version_compare('1.4.0', '>=', strict=True) }}"
+    - "{{ elasticsearch_installed_version.stdout | version_compare('1.6.9', '<=', strict=True) }}"
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - elasticsearch-pre-flight
+
+- debug:
+    msg: "ElasticSearch is at the latest version, skipping reindexing steps."
+  when: "{{ latest_version | bool }}"
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - elasticsearch-pre-flight
+
+- fail:
```

Comment:
@git-harry there is a way but I'm concerned about the scenario of a very long runing reindexing job.  It took several days to reindex the logs on OSIC1 and that is not a realistic time for the play to monitor.  If the upgrade play is run when elasticsearch is in the 1.7.x branch then the user needs to be absolutely sure what state the environment is in before continuing.